src/encoded/auth0.py

```diff
@@ -96,6 +97,54 @@ def forget(self, request):
         return []
 
 
+@view_config(route_name='signup', request_method='POST', permission=NO_PERMISSION_REQUIRED)
+def signup(request):
+    """
+    Create new user.
+
+    :param request: Pyramid request object
+    """
+    domain = 'encode.auth0.com'
+    access_token = request.json.get('accessToken', '')
+
+    if not access_token:
+        raise HTTPBadRequest(explanation='Access token required.')
+
+    url = "https://{domain}/userinfo?access_token={access_token}".format(domain=domain, access_token=access_token)
+    user_data_request = requests.get(url)
+
+    if user_data_request.status_code != 200:
+        raise HTTPBadRequest(explanation='Could not get user data')
+
+    user_data = user_data_request.json()
+
+    if user_data['email_verified'] is not True:
+        raise HTTPBadRequest(explanation='Unverified e-mail')
+
+    if user_data.get('name', ''):
+        name_split = user_data.get('name', '').split(' ')
+        name_length = len(name_split)
+        name1 = name_split[0] if name_length >= 1 else user_data['email']
+        name2 = name_split[-1] if name_length >= 2 else user_data['email']
+
+    user_info = {
```

Comment:
You should try splitting your code out into functions that do exactly one thing and can be unit tested. Same goes for the name parsing logic above.